src/inference/src/dev/threading/thread_affinity.cpp

```diff
@@ -87,8 +87,9 @@ bool pin_thread_to_vacant_core(int thrIdx,
 }
 
 bool pin_current_thread_to_socket(int socket) {
-    const int sockets = ov::get_available_numa_nodes().size();
-    const int cores = ov::get_number_of_cpu_cores();
```

Comment:
You are not using `get_number_of_cpu_cores` anymore. Does it mean `get_number_of_cpu_cores` returns incorrect value on systems with more than 1024 cores?

Reply:
Yes, you are right. Because `sched_getaffinity` which returns incorrect value in machines of more than 1024 cores is called in `get_number_of_cpu_cores`. But I changed this api, it will return correct values. Consider `proc_type_table` includes all CPU information, so there is no need for `get_available_numa_nodes` and `get_number_of_cpu_cores` to be called here.